stdsimd/arch/detect/os/aarch64.rs

```diff
@@ -0,0 +1,64 @@
+//! Run-time feature detection for Aarch64 on any OS that traps the mrs instruction.
+
+use arch::detect::Feature;
+use arch::detect::cache;
+
+/// Try to read the features from the system registers.
+///
+/// This will cause SIGILL when the current OS is not trapping the mrs instruction.
+pub(crate) fn detect_features() -> cache::Initializer {
+    let mut value = cache::Initializer::default();
+
+    {
+        let mut enable_feature = |f, enable| {
+            if enable {
+                value.set(f as u32);
+            }
+        };
+
+        // ID_AA64ISAR0_EL1 - Instruction Set Attribute Register 0
+        let aa64isar0: u64;
+        unsafe { asm!("mrs $0, S3_0_C0_C6_0" : "=r"(aa64isar0)); }
```

Comment:
@japaric do we have an intrinsic for this or is one planned ?

Reply:
@gnzlbg That's the `rsr64` intrinsic added in the other open PR but it doesn't this form yet.

@myfreeweb Probably better to use `ID_AA64ISAR0_EL1` instead of  `S3_0_C0_C6_0` in the `asm!` to make it more readable/easier to verify.

Reply:
@myfreeweb see @parched suggestion to use `ID_AA64ISAR0_EL1` instead of `S3_0_C0_C6_0`.